File: eBordo.Api/Database/UtakmicaIzmjena.cs
﻿using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace eBordo.Api.Database
{
    public class UtakmicaIzmjena
    {
        public int utakmicaIzmjenaID { get; set; }

        public int utakmicaId { get; set; }
        public Utakmica utakmica { get; set; }

        public Igrac igracOut { get; set; }
        public int igracOutId { get; set; }

        public Igrac igracIn { get; set; }
        public int igracInId { get; set; }

        public int minuta { get; set; }
        public IzmjenaRazlog izmjenaRazlog { get; set; }
    }
}

File: ebordo_mobile/lib/models/utakmica-izmjena.dart
import 'igrac.dart';

class UtakmicaIzmjena {
  final Igrac igracOut;
  final Igrac igracIn;
  final int minuta;
  final num IzmjenaRazlog;

  UtakmicaIzmjena(
      {required this.igracOut,
      required this.igracIn,
      required this.minuta,
      required this.IzmjenaRazlog});

  factory UtakmicaIzmjena.fromJson(Map<String, dynamic> json) {
    return UtakmicaIzmjena(
      igracOut: Igrac.fromJson(json['igracOut']),
      igracIn: Igrac.fromJson(json['igracIn']),
      minuta: json["minuta"],
      IzmjenaRazlog: json["IzmjenaRazlog"],
    );
  }

  Map<String, dynamic> toJson() => {
        "igracOut": igracOut,
        "igracIn": igracIn,
        "minuta": minuta,
        "IzmjenaRazlog": IzmjenaRazlog
      };
}
